app/views/spree/admin/variants/_autocomplete.js.erb

```diff
@@ -16,16 +16,13 @@
       </ul>
       <ul class='variant-data'>
         <li class='variant-sku'><strong><%= t('admin.sku') %>:</strong> {{variant.sku}}</li>
-        <li class='variant-on_hand'><strong><%= t('on_hand') %>:</strong> {{variant.on_hand}}</li>
+        {{#if variant.on_demand }}
```

Comment:
I have no idea what this `#if ` syntax is and the same time it's a JS file with ERB in it? :astonished:

Reply:
this is an if clause in erb.
this is not a "JS file with erb" this is an "erb file with JS". the output of this erb file is a JS template (script type='text/template') that is injected by this complex JS code: https://github.com/openfoodfoundation/spree/blob/2-0-4-stable/backend/app/assets/javascripts/admin/variant_autocomplete.js.erb

this should be a haml template under app/assets that angular code inject...
this PR solved this issue #3440 partially but the JS code still needs to be changed, but it's complicated, see the link above...